### PR TITLE
Add option for setting Kubernetes feature gates

### PR DESCRIPTION
**Pull request by hakman**
/cc @olemarkus @rifelpet @jsafrane

### CODE REVIEW

**Comment by hakman** [comment]
Managed to test this with multiple k8s feature gates. No issues noticed.
/cc @johngmyers

**Comment by johngmyers** [comment]
@k8s-ci-robot needs-rebase

File: upup/pkg/fi/cloudup/new_cluster.go [hunk 1]
```diff
@@ -64,6 +63,8 @@ type NewClusterOptions struct {
 	DiscoveryStore string
 	// KubernetesVersion is the version of Kubernetes to deploy. It defaults to the version recommended by the channel.
 	KubernetesVersion string
+	// KubernetesVersion is the version of Kubernetes to deploy. It defaults to the version recommended by the channel.
```
Comment (johngmyers) [inline upup/pkg/fi/cloudup/new_cluster.go:66]: Incorrect comment

File: upup/pkg/fi/cloudup/new_cluster.go [hunk 2]
```diff
@@ -233,9 +285,11 @@ func NewCluster(opt *NewClusterOptions, clientset simple.Clientset) (*NewCluster
 	cluster.Spec.IAM = &api.IAMSpec{
 		AllowContainerRegistry: true,
 	}
-	cluster.Spec.Kubelet = &api.KubeletConfigSpec{
-		AnonymousAuth: fi.PtrTo(false),
+
+	if cluster.Spec.Kubelet == nil {
+		cluster.Spec.Kubelet = &api.KubeletConfigSpec{}
```
Comment (johngmyers) [inline upup/pkg/fi/cloudup/new_cluster.go:290]: It would be simpler to initialize `cluster.Spec.Kubelet` before the iteration over KubernetesFeatureGates

**Review by johngmyers — CHANGES_REQUESTED** [review]
Please fix the comment.

**Comment by johngmyers** [comment]
I wonder if we should put `KubernetesFeatureGates` in the cluster spec. It would be a lot easier to manage changes that way.

File: upup/pkg/fi/cloudup/new_cluster.go (hunk 2 above)
Comment (hakman) [reply upup/pkg/fi/cloudup/new_cluster.go]: Should be simpler now.

File: upup/pkg/fi/cloudup/new_cluster.go (hunk 1 above)
Comment (hakman) [reply upup/pkg/fi/cloudup/new_cluster.go]: Fixed.

**Comment by hakman** [comment]
> I wonder if we should put `KubernetesFeatureGates` in the cluster spec. It would be a lot easier to manage changes that way.

For sure would be easier. Was thinking that it would be nice in v1alpha3 API as it would mean removing individual flags from each component.

**Comment by johngmyers** [comment]
I don't think we can just remove the individual flags in v1alpha3. I was thinking of having the individual flags be merged on top of the cluster-level ones. Or we could have an API validation that says you can't set both.

**Comment by hakman** [comment]
Not sure if that will help that much. I think I am ok with how it is for now.

**Review by johngmyers — APPROVED** [review]

**Pull request merged**